### PR TITLE
EZP-26044: Handle height change and scroll on navigation

### PR DESCRIPTION
**Pull request by StephaneDiot**
jira : https://jira.ez.no/browse/EZP-26044 and https://jira.ez.no/browse/EZP-26470

## Description

The first goal of this fix was to correct the bug where navigation hub was hiding upper part of the screen. After investigation the opposite effect (screen was moved down) was also found in some cases.
To fix those effects I had to correct https://jira.ez.no/browse/EZP-26470 (which is basically that navigating between diferent location view were keeping the scroll offset instead of reseting it). So this pull request is fixing both https://jira.ez.no/browse/EZP-26044 and https://jira.ez.no/browse/EZP-26470

## Tests

Unit and manually tested

### CODE REVIEW

File: Resources/public/js/views/ez-locationviewview.js [hunk 1]
```diff
@@ -31,6 +31,7 @@ YUI.add('ez-locationviewview', function (Y) {
                 Y.Array.each(this.get('tabs'), function (t) {
                     t.set('active', this.get('active'));
                 }, this);
+                window.scrollTo(0,0);
```
Comment (dpobel) [inline Resources/public/js/views/ez-locationviewview.js]: as far as I get it, it's not the right way (tm) to fix the scroll issue. First, this should be made once and for all views not only the LocationView. Second, you should probably take example on the YUI code doing that (see https://yuilibrary.com/yui/docs/api/files/pjax_js_pjax-base.js.html#l305)

**Comment by StephaneDiot** [comment]
updated @dpobel

File: Tests/js/apps/assets/ez-platformuiapp-tests.js [hunk 2]
```diff
@@ -199,6 +199,37 @@ YUI.add('ez-platformuiapp-tests', function (Y) {
...
+        },
+
+        tearDown: function () {
+            
+            this.app.destroy();
+            delete this.app;
+        },
+
+        "Should scroll to top and left of the window when showing main view": function () {
+            var that = this;
```
Comment (dpobel) [inline Tests/js/apps/assets/ez-platformuiapp-tests.js]: please no more `that` :)

File: Tests/js/apps/assets/ez-platformuiapp-tests.js [hunk 3]
```diff
@@ -199,6 +199,37 @@ YUI.add('ez-platformuiapp-tests', function (Y) {
...
+
+        setUp: function () {
+            this.app = new Y.eZ.PlatformUIApp({
+                container: '.app',
+                viewContainer: '.view-container'
+            });
+        },
+
+        tearDown: function () {
+
```
Comment (dpobel) [inline Tests/js/apps/assets/ez-platformuiapp-tests.js]: cs

File: Resources/public/js/apps/ez-platformuiapp.js [hunk 4]
```diff
@@ -736,6 +736,16 @@ YUI.add('ez-platformuiapp', function (Y) {
                         update: true,
                         render: true
                     });
+
+                    if (window && app.get('scrollToTop')) {
```
Comment (dpobel) [inline Resources/public/js/apps/ez-platformuiapp.js]: no need to test `window`, it's always there

**Comment by yannickroger** [comment]
Is there a reason why base branch is not 1.7?

File: Tests/js/views/assets/ez-locationviewview-tests.js [hunk 5]
```diff
@@ -1093,7 +1093,6 @@ YUI.add('ez-locationviewview-tests', function (Y) {
         },
     });
 
-
```
Comment (yannickroger) [inline Tests/js/views/assets/ez-locationviewview-tests.js]: nitpick, not needed.

**Pull request merged**